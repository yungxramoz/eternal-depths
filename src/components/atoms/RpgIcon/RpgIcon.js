import React from 'react'

const RpgIcon = ({ icon, className = '' }) => {
  return <div className={`rpgui-icon ${icon} ${className}`}></div>
}

export default RpgIcon
