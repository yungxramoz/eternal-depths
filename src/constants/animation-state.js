const IDLE_STANDING = 'animate-idle-standing'
const IDLE_FLYING = 'animate-idle-flying'
const ATTACKING = 'animate-attacking'
const DAMAGING = 'animate-damaging'

export const ANIMATION_STATE = {
  IDLE_STANDING,
  IDLE_FLYING,
  ATTACKING,
  DAMAGING,
}
