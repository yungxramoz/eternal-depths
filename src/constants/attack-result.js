const HIT = 'hit'
const CRITICAL = 'critical'
const EVADED = 'evaded'
const MISSED = 'missed'

const ATTACK_RESULT = {
  HIT,
  CRITICAL,
  EVADED,
  MISSED,
}

export default ATTACK_RESULT
