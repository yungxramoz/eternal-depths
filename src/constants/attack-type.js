import STAT from './stat'

const NORMAL = 'normal'

export const ATTACK_TYPE = {
  NORMAL,
}

export const EMPTY_ATTACK = Object.freeze({
  name: 'Empty Attack',
  description: 'An empty attack.',
  icon: 'attack-empty',
  type: 'normal',
  selfHealAmount: 0,
  selfInflictedAmount: 0,
  hitCount: 1,
  additionalDamage: 0,
  damageDecrease: 0,
  criticalChance: 0,
  cooldown: 0,
  buffs: [],
})

export const BASE_ATTACK = Object.freeze({
  name: 'Base Attack',
  description: 'A basic attack that deals damage to the enemy.',
  icon: 'attack-base',
  type: 'normal',
  selfHealAmount: 0,
  selfInflictedAmount: 0,
  hitCount: 1,
  additionalDamage: 0,
  damageDecrease: 0,
  criticalChance: 0,
  cooldown: 0,
  buffs: [],
})

const DRAIN = Object.freeze({
  name: 'Drain',
  description: 'Steal health from the enemy.',
  icon: 'attack-drain',
  type: 'normal',
  selfHealAmount: 'auto',
  selfInflictedAmount: 0,
  hitCount: 1,
  additionalDamage: 0,
  criticalChance: 0,
  cooldown: 2,
  buffs: [],
})

const DANCE_OF_THE_DEAD = Object.freeze({
  name: 'Dance of the Dead',
  description: 'Attack the enemy multiple times and increase agility.',
  icon: 'attack-dance-of-the-dead',
  type: 'normal',
  selfHealAmount: 0,
  selfInflictedAmount: 0,
  hitCount: 3,
  additionalDamage: 0,
  criticalChance: 0,
  cooldown: 2,
  buffs: [
    {
      stat: STAT.AGILITY,
      value: 5,
      duration: 1,
    },
  ],
})

const BLOODLUST = Object.freeze({
  name: 'Bloodlust',
  description: 'Increase damage and critical chance.',
  icon: 'attack-bloodlust',
  type: 'normal',
  selfHealAmount: 0,
  selfInflictedAmount: 0,
  hitCount: 1,
  cooldown: 2,
  additionalDamage: 0,
  buffs: [
    {
      stat: STAT.STRENGTH,
      value: 5,
      duration: 1,
    },
    {
      stat: STAT.PRECISION,
      value: 5,
      duration: 1,
    },
  ],
})

const BERSERK_STRIKE = Object.freeze({
  name: 'Berserk Strike',
  description: 'Deal massive damage to the enemy and inflicts damage to self.',
  icon: 'attack-berserk-strike',
  type: 'normal',
  selfHealAmount: 0,
  selfInflictedAmount: 6,
  hitCount: 1,
  additionalDamage: 20,
  criticalChance: 0,
  cooldown: 3,
  buffs: [],
})

const DOUBLE_STRIKE = Object.freeze({
  name: 'Double Strike',
  description: 'Attack the enemy twice.',
  icon: 'attack-double-strike',
  type: 'normal',
  selfHealAmount: 0,
  selfInflictedAmount: 0,
  hitCount: 2,
  additionalDamage: 0,
  criticalChance: 0,
  cooldown: 1,
  buffs: [],
})

const DEATH_BLOW = Object.freeze({
  name: 'Death Blow',
  description: 'Deal massive damage to the enemy.',
  icon: 'attack-death-blow',
  type: 'normal',
  selfHealAmount: 0,
  selfInflictedAmount: 0,
  hitCount: 1,
  additionalDamage: 50,
  criticalChance: 0,
  cooldown: 5,
  buffs: [],
})

export const ATTACK = {
  DRAIN,
  DANCE_OF_THE_DEAD,
  BLOODLUST,
  BERSERK_STRIKE,
  DOUBLE_STRIKE,
  DEATH_BLOW,
}
