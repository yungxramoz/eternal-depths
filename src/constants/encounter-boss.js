export const FIRE_DRAGON = Object.freeze({
  name: 'Fire Dragon',
  fileName: 'fire-dragon.png',
  baseStats: {
    health: 3,
    strength: 3,
    agility: 2,
    precision: 2,
  },
  style: {
    width: '50vw',
    top: '0px',
    left: '0px',
  },
})

export const LICH = Object.freeze({
  name: 'Lich',
  fileName: 'lich.png',
  baseStats: {
    health: 2,
    strength: 2,
    agility: 3,
    precision: 3,
  },
  style: {
    width: '50vw',
    top: '0px',
    left: '0px',
  },
})

export const MAD_MAGE = Object.freeze({
  name: 'Mad Mage',
  fileName: 'mad-mage.png',
  baseStats: {
    health: 2,
    strength: 3,
    agility: 2,
    precision: 3,
  },
  style: {
    width: '50vw',
    top: '0px',
    left: '0px',
  },
})

export const SHADOW_DRAGON = Object.freeze({
  name: 'Shadow Dragon',
  fileName: 'shadow-dragon.png',
  baseStats: {
    health: 2,
    strength: 4,
    agility: 2,
    precision: 2,
  },
  style: {
    width: '50vw',
    top: '0px',
    left: '0px',
  },
})

export const VAMPIRE_LORD = Object.freeze({
  name: 'Vampire Lord',
  fileName: 'vampire-lord.png',
  baseStats: {
    health: 3,
    strength: 2,
    agility: 3,
    precision: 3,
  },
  style: {
    width: '50vw',
    top: '0px',
    left: '0px',
  },
})

export const BossEncounters = [
  FIRE_DRAGON,
  LICH,
  MAD_MAGE,
  SHADOW_DRAGON,
  VAMPIRE_LORD,
]
