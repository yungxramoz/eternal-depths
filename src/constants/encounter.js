export const BARBARIAN = Object.freeze({
  name: 'Barbarian',
  fileName: 'barbarian.png',
  stages: ['dungeon-3.png', 'dungeon-4.png', 'dungeon-5.png'],
  baseStats: {
    health: 2,
    strength: 2,
    agility: 1,
    precision: 1,
  },
  style: {
    maxWidth: '60vw',
    maxHeight: 'd60vh',
    alignSelf: 'flex-end',
  },
})

export const BAT = Object.freeze({
  name: 'Giant Bat',
  fileName: 'bat.png',
  stages: ['cave-0.png', 'cave-1.png', 'cave-2.png', 'cave-3.png'],
  baseStats: {
    health: 1,
    strength: 1,
    agility: 3,
    precision: 1,
  },
  style: {
    maxWidth: '70vw',
    maxHeight: '70dvh',
    alignSelf: 'flex-start',
  },
})

export const GOBLIN = Object.freeze({
  name: 'Goblin',
  fileName: 'goblin.png',
  stages: ['dungeon-0.png', 'dungeon-1.png', 'dungeon-2.png'],
  baseStats: {
    health: 1,
    strength: 2,
    agility: 2,
    precision: 1,
  },
  style: {
    maxWidth: '45vw',
    maxHeight: '45dvh',
    alignSelf: 'flex-end',
  },
})

export const ORC = Object.freeze({
  name: 'Orc',
  fileName: 'orc.png',
  stages: ['dungeon-3.png', 'dungeon-5.png'],
  baseStats: {
    health: 3,
    strength: 2,
    agility: 0,
    precision: 0,
  },
  style: {
    maxWidth: '70vw',
    maxHeight: '70dvh',
    alignSelf: 'flex-end',
  },
})

export const SCOUNDREL = Object.freeze({
  name: 'Scoundrel',
  fileName: 'scoundrel.png',
  stages: ['dungeon-3.png', 'dungeon-4.png', 'dungeon-5.png'],
  baseStats: {
    health: 1,
    strength: 1,
    agility: 1,
    precision: 1,
  },
  style: {
    maxWidth: '60vw',
    maxHeight: '60dvh',
    alignSelf: 'flex-end',
  },
})

export const SKELETON = Object.freeze({
  name: 'Skeleton',
  fileName: 'skeleton.png',
  stages: ['dungeon-4.png', 'dungeon-5.png'],
  baseStats: {
    health: 2,
    strength: 1,
    agility: 1,
    precision: 0,
  },
  style: {
    maxWidth: '60vw',
    maxHeight: '60dvh',
    alignSelf: 'flex-end',
  },
})

export const SNAKE = Object.freeze({
  name: 'Snake',
  fileName: 'snake.png',
  stages: ['cave-1.png', 'cave-2.png', 'cave-3.png'],
  baseStats: {
    health: 1,
    strength: 1,
    agility: 2,
    precision: 2,
  },
  style: {
    maxWidth: '40vw',
    maxHeight: '40dvh',
    alignSelf: 'flex-end',
  },
})

export const SPIDER = Object.freeze({
  name: 'Spider',
  fileName: 'spider.png',
  stages: ['cave-1.png', 'cave-2.png', 'dungeon-4.png', 'dungeon-5.png'],
  baseStats: {
    health: 1,
    strength: 1,
    agility: 2,
    precision: 1,
  },
  style: {
    maxWidth: '60vw',
    maxHeight: '60dvh',
    alignSelf: 'flex-end',
  },
})

export const THIEF = Object.freeze({
  name: 'Thief',
  fileName: 'thief.png',
  stages: ['dungeon-3.png', 'dungeon-4.png', 'dungeon-5.png'],
  baseStats: {
    health: 1,
    strength: 0,
    agility: 2,
    precision: 3,
  },
  style: {
    maxWidth: '60vw',
    maxHeight: '60dvh',
    alignSelf: 'flex-end',
  },
})

export const WOLF = Object.freeze({
  name: 'Wolf',
  fileName: 'wolf.png',
  stages: ['dungeon-3.png', 'cave-5.png'],
  baseStats: {
    health: 2,
    strength: 2,
    agility: 1,
    precision: 0,
  },
  style: {
    maxWidth: '50vw',
    maxHeight: '50dvh',
    alignSelf: 'flex-end',
  },
})

export const ENCOUNTERS = [
  BARBARIAN,
  BAT,
  GOBLIN,
  ORC,
  SCOUNDREL,
  SKELETON,
  SNAKE,
  SPIDER,
  THIEF,
  WOLF,
]
