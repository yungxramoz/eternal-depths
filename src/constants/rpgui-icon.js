const SWORD = 'sword'
const RUSTY_SWORD = 'rusty-sword'
const SHIELD = 'shield'
const HELMET = 'helmet'
const ARMOR = 'armor'
const SHOES = 'shoes'
const GLOVES = 'gloves'
const FOOD = 'food'
const BOOK = 'book'
const POTION = 'potion'
const SKULL = 'skull'
const TORCH = 'torch'
const TREASURE = 'treasure'

export const RPGUI_ICON = {
  SWORD,
  RUSTY_SWORD,
  SHIELD,
  HELMET,
  ARMOR,
  SHOES,
  GLOVES,
  FOOD,
  BOOK,
  POTION,
  SKULL,
  TORCH,
  TREASURE,
}
