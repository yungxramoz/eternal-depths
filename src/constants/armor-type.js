import { RPGUI_ICON } from './rpgui-icon'

export const HELMET = Object.freeze({
  type: 'helmet',
  icon: RPGUI_ICON.HELMET,
  stats: {
    health: 0,
    precision: 0,
  },
})
export const ARMOR = Object.freeze({
  type: 'armor',
  icon: RPGUI_ICON.ARMOR,
  stats: {
    health: 0,
    strength: 0,
    agility: 0,
  },
})
export const GLOVES = Object.freeze({
  type: 'gloves',
  icon: RPGUI_ICON.GLOVES,
  stats: {
    strength: 0,
    agility: 0,
    precision: 0,
  },
})
export const GREAVES = Object.freeze({
  type: 'greaves',
  icon: RPGUI_ICON.SHOES,
  stats: {
    agility: 0,
    precision: 0,
  },
})

const ARMOR_TYPE = {
  HELMET,
  ARMOR,
  GLOVES,
  GREAVES,
}

export const ARMOR_TYPES = Object.values(ARMOR_TYPE)

export default ARMOR_TYPE
