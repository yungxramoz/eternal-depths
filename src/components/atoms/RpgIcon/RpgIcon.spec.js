import { render } from '@testing-library/react'
import React from 'react'
import { RPGUI_ICON } from '../../../constants/rpgui-icon'
import RpgIcon from './RpgIcon'

describe('RpgIcon', () => {
  it('renders correctly', () => {
    render(<RpgIcon icon={RPGUI_ICON.SWORD} />)
    expect(document.body).toMatchSnapshot('RpgIcon')
  })
})
