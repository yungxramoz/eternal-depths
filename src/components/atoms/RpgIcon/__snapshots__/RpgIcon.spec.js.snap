// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RpgIcon renders correctly: RpgIcon 1`] = `
<body>
  <div>
    <div
      class="rpgui-icon sword "
    />
  </div>
</body>
`;
